### PR TITLE
Add \GdImage class

### PR DESCRIPTION
**Pull request by b1rdex**
https://php.watch/versions/8.0/gdimage

### CODE REVIEW

File: gd/gd.php [hunk 1]
```diff
@@ -3013,3 +3013,8 @@ function imagescale ($image, $new_width, $new_height = -1, $mode = IMG_BILINEAR_
  * @since 5.5
  */
 function imagesetinterpolation ($image, $method = IMG_BILINEAR_FIXED) {}
+
+/**
+ * @since 8.0
+ */
+final class GdImage{}
```
Comment (MaXal) [inline gd/gd.php:3020]: Should we add a private constructor to enforce: `Cannot be instantiated with new \GdImage().` ?

**Review by MaXal — CHANGES_REQUESTED** [review]

File: gd/gd.php (hunk 1 above)
Comment (b1rdex) [reply gd/gd.php:3020]: Decide yourself. I did it the same as `CurlHandler` – just a class definition.

Comment (b1rdex) [reply gd/gd.php:3020]: FYI: they aren't cloneable too: https://github.com/php/php-src/blob/master/ext/gd/tests/gdimage_prevent_cloning.phpt

Comment (b1rdex) [reply gd/gd.php:3020]: But CurlHandle *is* cloneable: https://github.com/php/php-src/blob/master/ext/curl/tests/curl_handle_clone.phpt (sure why not 🤣)

**Pull request merged**